File: C:C++/Queue.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Node
{
    int data;
    Node *next;
}Node;

typedef struct Queue
{
    Node *top;
    Node *tail;
}Queue;

Queue *createQueue()
{
    Queue *Q;
    Q = (Queue *)malloc(sizeof(Queue));
    Q->top = NULL;
    Q->tail = NULL;
    return Q;
}

int Dequeue(Queue *Q)
{
    if (empty(Q))
        printf("Queue is Empty\n");  
    else
    {
        Node *temp = malloc(sizeof(Node));
        temp = Q->top;
        int num = temp->data;
        Q->top = temp->next;
        free(temp);
        Q->front++;
        return num;
    }    
}

void Enqueue(Queue *Q, int element)
{
    Node *temp = malloc(sizeof(Node));
    temp->data = element;
    temp->next = NULL;
    Q->tail->next = temp;
    Q->tail = temp;    
}

int Front(Queue *Q)
{
    if (empty(Q))
        printf("Queue is Empty\n");
    else
        return Q->elements[Q->head];
}

bool empty(Queue *Q) {
    
    return (Q->top == NULL);
}

File: Java/Queue.java
public class Queue<Item> {

	private Node top;
	private Node tail;
    private int N;
    
    private class Node {
        Item item;
        Node next;
    }

	public Queue() {
		this.top = null;
		this.tail = null;
        this.N = 0;
	}	

	public Item dequeue() {
        if(this.isEmpty())
            return null;
        Node temp = this.top;
        Item num = temp.val;
        this.top = temp.next;
        N++;
        return Item;
	}

	public void enqueue(Item element) {
		Node temp = new Node();
        temp.item = element;
		this.tail.next = temp;
		this.tail = temp;
        N--;
	}

	public boolean isEmpty() {
		return (this.top == null);
	}
    
    public int size() {
        return N;
    }


}
